src/swift/Block.swift

```diff
@@ -104,8 +104,8 @@ public class DispatchWorkItem {
 /// uses @convention(block). This is to avoid unnecessary bridging between
 /// C blocks and Swift closures, which interferes with dispatch APIs that depend
 /// on the referential identity of a block. Particularly, dispatch_block_create.
-internal typealias _DispatchBlock = @convention(block) () -> Void
-internal typealias dispatch_block_t = @convention(block) () -> Void
+internal typealias _DispatchBlock = @escaping @convention(block) () -> Void
+internal typealias dispatch_block_t = @escaping @convention(block) () -> Void
```

Comment:
@slavapestov ok, understood! I'm thinking on adding the following macro declaration to match the already existing `DISPATCH_NOESCAPE` solution existing in [base.h](https://github.com/apple/swift-corelibs-libdispatch/blob/ab16f5e62859ff2f54996b8838f8304a8d125102/dispatch/base.h#L164), and keep compatibility with different compiler versions/implementations:

```
#if __has_attribute(escaping)
#define DISPATCH_ESCAPING __attribute__((__escaping__))
#define DISPATCH_NOESCAPE
#if __has_attribute(noescape)
#define DISPATCH_ESCAPING
#define DISPATCH_NOESCAPE __attribute__((__noescape__))
#else
#define DISPATCH_ESCAPING
#define DISPATCH_NOESCAPE
#endif
```

@mwwa / @MadCoder I think this answers the question. I'm gonna keep working on this, unless someone is already doing it internally.

Thanks again for the comments.